components/camel-amqp/src/test/java/org/apache/camel/component/amqp/AMQPRouteTest.java

```diff
@@ -26,29 +26,49 @@
 import jakarta.jms.TextMessage;
 
 import org.apache.camel.CamelContext;
-import org.apache.camel.EndpointInject;
+import org.apache.camel.ProducerTemplate;
 import org.apache.camel.RuntimeCamelException;
 import org.apache.camel.builder.RouteBuilder;
 import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.infra.artemis.services.ArtemisService;
+import org.apache.camel.test.infra.artemis.services.ArtemisServiceFactory;
+import org.apache.camel.test.infra.core.CamelContextExtension;
+import org.apache.camel.test.infra.core.ContextFixture;
+import org.apache.camel.test.infra.core.DefaultCamelContextExtension;
+import org.apache.camel.test.infra.core.RouteFixture;
 import org.apache.qpid.jms.message.JmsMessage;
 import org.apache.qpid.jms.provider.amqp.message.AmqpJmsMessageFacade;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 import static org.apache.camel.component.amqp.AMQPComponent.amqpComponent;
 import static org.apache.camel.component.amqp.AMQPConnectionDetails.discoverAMQP;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
-public class AMQPRouteTest extends AMQPTestSupport {
+public class AMQPRouteTest {
 
-    @EndpointInject("mock:result")
-    MockEndpoint resultEndpoint;
+    @RegisterExtension
+    protected static ArtemisService service = ArtemisServiceFactory.createSingletonAMQPService();
 
-    String expectedBody = "Hello there!";
+    @RegisterExtension
+    protected static CamelContextExtension contextExtension = new DefaultCamelContextExtension();
```

Comment:
Can this extension be added to AMQPTestSupport class? just to avoid duplicates

Reply:
Absolutely! Let me do this change :)

Reply:
Done :)